dashboard/test/helpers/delete_accounts_helper_test.rb

```diff
@@ -238,6 +238,21 @@ class DeleteAccountsHelperTest < ActionView::TestCase
     refute_nil user_geo.country
   end
 
+  test 'does not purge dependent students of a teacher' do
+    student = create :student_in_picture_section
+    teacher = student.teachers.first
+    assert_includes teacher.dependent_students.map {|s| s[:id]}, student.id
+
+    assert_nil teacher.purged_at
+    assert_nil student.purged_at
+
+    purge_user teacher
+
+    student.reload
+    refute_nil teacher.purged_at
+    assert_nil student.purged_at
+  end
```

Comment:
Verified (red-green testing) that this test case fails if our code purges the student, whether the additional logic in the `purge_user` helper below is present or not.  In the final state the `purge_user` helper check, which applies to all tests in this file, is the first failure in this test, which gives the most helpful output.